pinot-core/src/main/java/org/apache/pinot/core/operator/CombineOperator.java

```diff
@@ -196,6 +202,17 @@ public IntermediateResultsBlock callJob()
     return mergedBlock;
   }
 
+  // This will check if IntermediateResultsBlock already satisfying query, so there is no need to continue query processing.
+  private boolean allowEarlyTermination(BrokerRequest brokerRequest, IntermediateResultsBlock mergedBlock) {
```

Comment:
Can we rename this method to something more intuitive? Like `isQuerySatisfied()`?